tests/overmap_test.cpp

```diff
@@ -19,3 +20,21 @@ TEST_CASE( "set_and_get_overmap_scents" ) {
     REQUIRE( test_overmap.scent_at( { 75, 85, 0} ).creation_turn == 50 );
     REQUIRE( test_overmap.scent_at( { 75, 85, 0} ).initial_strength == 90 );
 }
+
+TEST_CASE( "default_overmap_generation_always_succeeds" ) {
+    // List of specials to generate, overmap should consume them all.
+    std::vector<const overmap_special *> default_specials = overmap( 0, 0 ).get_enabled_specials();
+    for( int i = 1; i < 10; i++ ) {
+        std::vector<overmap_special_placement> test_specials;
+        for( auto special : default_specials ) {
+            test_specials.push_back( { 0, special } );
+        }
+        overmap_buffer.create_custom_overmap( i, i, test_specials );
```

Comment:
Only diagonals? That's an easy case.
You use nearest points in radius when deferring specials. Using that here would be more likely to catch problems.